backend/modules/orders/order_engine/engine.py

```diff
@@ -99,9 +100,42 @@ def _process_order_group(self):
                 continue
             if isinstance(o, MarketEngineOrder):
                 self._handle_market_order(o)
+            elif isinstance(o, LimitEngineOrder):
+                self._handle_limit_order(o)
             else:
                 raise RuntimeError("Unsupported order type")
 
+    def _handle_limit_order(self, order: LimitEngineOrder):
+        """
+        Execute a limit order only when the market price meets the limit price
+        condition. For buy orders we require market_price <= limit_price.
+        For sell orders we require market_price >= limit_price.
+        Otherwise the order stays untouched.
+        """
+        ticker = order.ticker
+        price = self._prices[ticker]
+        market_order = MarketEngineOrder(
+            id=order.id,
+            ticker=order.ticker,
+            investor_id=order.investor_id,
+            volume=order.volume,
+            is_buy=order.is_buy,
+            volume_processed=order.volume_processed,
+        )
+
+        # Check if limit condition is satisfied
+        if order.is_buy:
+            # buy only when market price is at or below limit price
+            if price > order.limit_price:
+                return
+            # then behave like market buy at current market price
+            self._handle_market_buy(market_order)
```

Comment:
Nice trick, buy you can't just pass MarketOrder to _handle_market_buy, as it will append MarketOrderUpdate to updated orders. You can either rename MarketOrderUpdate to OrderUpdate and treat it as such (it seems that 'volume_processed' will always be updated) or make it create LimitOrderUpdate